dateparser/search/search.py

```diff
@@ -171,27 +171,37 @@ def __init__(self):
         self.available_language_map = self.loader.get_locale_map()
         self.search = _ExactLanguageSearch(self.loader)
 
-    def detect_language(self, text, languages):
-        if isinstance(languages, (list, tuple, Set)):
+    @apply_settings
+    def detect_language(self, text, languages, settings=None, detect_languages_func=None):
+        if detect_languages_func:
+            detected_languages = detect_languages_func(text, settings=settings)
+            detected_languages = detected_languages or None
```

Comment:
```suggestion
            detected_languages = detect_languages_func(text, settings=settings) or None
```

Reply:
why do you need `detected_languages or None`? for the next line, if you do `if not detected_languages:` and `detected_languages` is an empty array it will also return True